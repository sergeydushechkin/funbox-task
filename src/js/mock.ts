const goods = [
  {
    id: 0,
    title: `с фуа-гра`,
    description: `Печень утки разварная с артишоками.`,
    weight: `0,5`,
    amount: 11,
    consist: [
      {
        name: `порций`,
        value: 10
      },
      {
        name: `мышь в подарок`,
        value: null
      },
    ]
  },
  {
    id: 1,
    title: `с рыбой`,
    description: `Головы щучьи с чесноком да свежайшая сёмгушка.`,
    weight: `2`,
    amount: 7,
    consist: [
      {
        name: `порций`,
        value: 40
      },
      {
        name: `мыши в подарок`,
        value: 2
      },
    ]
  },
  {
    id: 2,
    title: `с курой`,
    description: `Филе из цыплят с трюфелями в бульоне.`,
    weight: `5`,
    amount: 0,
    consist: [
      {
        name: `порций`,
        value: 100
      },
      {
        name: `мышей в подарок`,
        value: 5
      },
      {
        name: `заказчик доволен`,
        value: null
      },
    ]
  },
];

export {goods};
