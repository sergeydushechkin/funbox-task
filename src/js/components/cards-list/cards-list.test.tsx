import * as React from "react";
import * as renderer from "react-test-renderer";

import CardsList from "./cards-list";
import {goods} from "../../mock";

it(`Render CardsList component`, () => {
  const cardsListElement = renderer
    .create(
        <CardsList goods={goods} />
    )
    .toJSON();

  expect(cardsListElement).toMatchSnapshot();
});
