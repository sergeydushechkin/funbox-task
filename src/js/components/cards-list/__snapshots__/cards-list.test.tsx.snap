// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CardsList component 1`] = `
<ul
  className="cards__list"
>
  <li
    className="cards__item"
  >
    <article
      className="cards_card card"
    >
      <div
        className="card__poster poster poster--unselected"
        tabIndex={1}
      >
        <div
          className="poster__content"
        >
          <p
            className="poster__text"
          >
            <span
              className="poster__phrase1"
            >
              Сказочное заморское яство
            </span>
            <span
              className="poster__phrase2"
            >
              Котэ не одобряет?
            </span>
          </p>
          <h2
            className="poster__title"
          >
            Нямушка
          </h2>
          <p
            className="poster__taste"
          >
            с фуа-гра
          </p>
          <ul
            className="poster__consist consist"
          >
            <li
              className="consist__item"
            >
              <b>
                10
              </b>
               порций
            </li>
            <li
              className="consist__item"
            >
              мышь в подарок
            </li>
          </ul>
        </div>
        <div
          className="poster__weight weight"
        >
          <span
            className="weight__value"
          >
            0,5
          </span>
          <span
            className="weight__measure"
          >
            кг
          </span>
        </div>
      </div>
      <p
        className="card__status"
      >
        Чего сидишь? Порадуй котэ, 
        <a
          className="card__link"
          href=""
        >
          купи.
        </a>
      </p>
    </article>
  </li>
  <li
    className="cards__item"
  >
    <article
      className="cards_card card card--selected"
    >
      <div
        className="card__poster poster poster--selected"
        tabIndex={1}
      >
        <div
          className="poster__content"
        >
          <p
            className="poster__text"
          >
            <span
              className="poster__phrase1"
            >
              Сказочное заморское яство
            </span>
            <span
              className="poster__phrase2"
            >
              Котэ не одобряет?
            </span>
          </p>
          <h2
            className="poster__title"
          >
            Нямушка
          </h2>
          <p
            className="poster__taste"
          >
            с рыбой
          </p>
          <ul
            className="poster__consist consist"
          >
            <li
              className="consist__item"
            >
              <b>
                40
              </b>
               порций
            </li>
            <li
              className="consist__item"
            >
              <b>
                2
              </b>
               мыши в подарок
            </li>
          </ul>
        </div>
        <div
          className="poster__weight weight"
        >
          <span
            className="weight__value"
          >
            2
          </span>
          <span
            className="weight__measure"
          >
            кг
          </span>
        </div>
      </div>
      <p
        className="card__status"
      >
        Головы щучьи с чесноком да свежайшая сёмгушка.
      </p>
    </article>
  </li>
  <li
    className="cards__item"
  >
    <article
      className="cards_card card card--disabled"
    >
      <div
        className="card__poster poster poster--disabled"
        tabIndex={-1}
      >
        <div
          className="poster__content"
        >
          <p
            className="poster__text"
          >
            <span
              className="poster__phrase1"
            >
              Сказочное заморское яство
            </span>
            <span
              className="poster__phrase2"
            >
              Котэ не одобряет?
            </span>
          </p>
          <h2
            className="poster__title"
          >
            Нямушка
          </h2>
          <p
            className="poster__taste"
          >
            с курой
          </p>
          <ul
            className="poster__consist consist"
          >
            <li
              className="consist__item"
            >
              <b>
                100
              </b>
               порций
            </li>
            <li
              className="consist__item"
            >
              <b>
                5
              </b>
               мышей в подарок
            </li>
            <li
              className="consist__item"
            >
              заказчик доволен
            </li>
          </ul>
        </div>
        <div
          className="poster__weight weight"
        >
          <span
            className="weight__value"
          >
            5
          </span>
          <span
            className="weight__measure"
          >
            кг
          </span>
        </div>
      </div>
      <p
        className="card__status card__status--not-available"
      >
        Печалька, с курой закончился.
      </p>
    </article>
  </li>
</ul>
`;
